File: API/Entities/Category.cs
namespace API.Entities;

public class Category
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string? PictureUrl { get; set; }
    public string? PublicId { get; set; }
    public List<Product> Products { get; set; } = new();

    public Category()
    {
    }

    public Category(string name, string? pictureUrl)
    {
        Name = name;
        PictureUrl = pictureUrl;
    }
}

File: client/src/app/models/category.ts
import {ProductLite} from "./product";

export interface Category {
    id: number;
    name: string;
    pictureUrl?: string;
    publicId?: string;
}

export interface CategoryFull {
    id: number;
    name: string;
    pictureUrl?: string;
    products: ProductLite[]
}